src/reduxTypes.js

```diff
@@ -346,10 +346,14 @@ export type RealmState = {|
   +serverEmojiData: ServerEmojiData | null,
 |};
 
-// TODO: Stop using the 'default' name. Any 'default' semantics should
-// only apply the device level, not within the app. See
-// https://github.com/zulip/zulip-mobile/issues/4009#issuecomment-619280681.
-export type ThemeName = 'default' | 'night';
+/**
+ * The theme of the app.
+ *
+ * * auto: follow the OS theme setting
+ * * light
+ * * dark
+ */
+export type ThemeName = 'auto' | 'light' | 'dark';
```

Comment:
It may be helpful to have two different types here:
* `ThemeName` for the name of a concrete specific theme -- either "light", or "dark";
* `ThemeSetting`, say, for the value the user has chosen -- so either a `ThemeName`, or "auto".

Before this feature, that distinction doesn't really exist (or if it does, it's pretty abstract) because the user's choices correspond 1:1 to specific themes. But with this PR, we have some places that want a value which could be "auto" but also a bunch of places that really want only a concrete "light" or "dark". Being able to express the difference will let the type-checker help us keep track and make sure we don't accidentally let the value "auto" flow to some code that assumes it's seeing only "light" or "dark".

Reply:
Done. Thanks for this idea. I think it greatly helps to differentiate where in the app it is ok to have the setting (auto, light, dark), vs what we really want in some places (only light or dark).